packages/amplify-graphql-api-construct/src/internal/data-source-config.ts

```diff
@@ -93,3 +99,65 @@ export const getDataSourceStrategiesProvider = (definition: IAmplifyGraphqlDefin
 
   return provider;
 };
+
+/**
+ * Creates a new schema by merging the individual schemas contained in the definitions, combining fields of the Query and Mutation types in
+ * individual definitions into a single combined definition. Adding directives to `Query` and `Mutation` types participating in a
+ * combination is not supported (the behavior is undefined whether those directives are migrated).
+ */
+export const schemaByMergingDefinitions = (definitions: IAmplifyGraphqlDefinition[]): string => {
+  const schema = definitions.map((def) => def.schema).join('\n');
+  const parsedSchema = parse(schema);
+
+  // We store the Query & Mutation definitions separately. Since the interfaces are readonly, we'll have to re-compose the types after we've
+  // collected all the fields
+  const queryAndMutationDefinitions: Record<
+    string,
+    {
+      node: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode;
+      fields: FieldDefinitionNode[];
+    }
+  > = {};
+
+  // Transform the schema by reducing Mutation & Query types:
+  // - Collect Mutation and Query definitions
+  // - Alter the parsed schema by filtering out Mutation & Query types
+  // - Add the combined Mutation & Query definitions to the filtered schema
+  parsedSchema.definitions.filter(isBuiltInGraphqlNode).forEach((def) => {
+    const typeName = def.name.value;
+    if (!queryAndMutationDefinitions[typeName]) {
+      queryAndMutationDefinitions[typeName] = {
+        node: def,
+        // `ObjectTypeDefinitionNode.fields` is a ReadonlyArray; so we have to create a new mutable array to collect all the fields
+        fields: [...(def.fields ?? [])],
+      };
+      return;
+    }
+
+    queryAndMutationDefinitions[typeName].fields = [...queryAndMutationDefinitions[typeName].fields, ...(def.fields ?? [])];
```

Comment:
If there is a custom SQL query field with same name in multiple schemas, this looks like we're overwriting them instead of throwing error?

Reply:
Validation added in [76cc531](https://github.com/aws-amplify/amplify-category-api/pull/2115/commits/76cc53150ab22ea2148ae3cfaa4b6563dfc20e2d)

Reply:
Sounds like this is the answer to part of my question above. Does our validation cover all possible naming conflicts?

Is the behavior documented?